lib/rspec/mocks/method_double.rb

```diff
@@ -102,10 +109,10 @@ def show_frozen_warning
 
       # @private
       def restore_original_visibility
-        return unless @original_visibility &&
-          MethodReference.method_defined_at_any_visibility?(object_singleton_class, @method_name)
+        return unless method_owner && @original_visibility &&
+          MethodReference.method_defined_at_any_visibility?(method_owner, @method_name)
 
-        object_singleton_class.__send__(@original_visibility, method_name)
+        method_owner.__send__(@original_visibility, @method_name)
```

Comment:
This method calls `method_owner` 3 different times.  The value shouldn't change between calls, and it seems inefficient to do that logic 3 times.  Can you store the value of `method_owner` in a variable so it's called only once?

Reply:
I updated `method_owner` to memorize as `@method_owner` now.